ArraysAndLists/Program.cs

```diff
@@ -19,49 +16,99 @@ public static class ArraysAndLists
+
+            var distinct = numbers.Distinct().Count();
+
+            var input = numbers.Length;
+
+           if (distinct != input)
+
+                return "Error";
+            else
+                Array.Sort(numbers);
+             
+            foreach (var number in numbers)
+            
+              result += number + " ";
+            
+            return result.Trim();
         }
 
         // 3 - Write a method that accepts a string for a name. Use an array to reverse the characters
         // then return the result as a string. "I like trains" should return "sniart ekil I". Return
         // "Error" if no text is inputted.
         public static string ReverseName(string name)
         {
-            // Delete the below line when you are writing your code, don't forget to use your
-            // own return statement.
-            return default;
+            var result = string.Empty;
+
+            var length = name.Length;
+            if (length != 0)
+                for (var i = length - 1; i >= 0; i--) 
+            result += name[i];
+            else 
+                return "Error";
+            return result;
+
         }
 
         // 4 - Write a method that accepts any number of integers. If none or less than 5 numbers,
         // return "Invalid Array"; otherwise, return the 3 smallest numbers as a string separated by
         // spaces, sorted in ascending order. Treat duplicate numbers normally.
         public static string SmallestNumbers(params int[] numbers)
         {
-            // Delete the below line when you are writing your code, don't forget to use your
-            // own return statement.
-            return default;
+         var result = string.Empty;
+            Array.Sort(numbers);
+            result = numbers.Length < 5 ? "Invalid Array" : $"{numbers[0]} {numbers[1]} {numbers[2]}";
+            
+            return result;
         }
 
         // 5 - Write a method that accepts any number of numbers (assume at least two numbers). The numbers
         // may include duplicates. Return the distinct numbers as an array sorted by ascending order.
         // So for example, an array consisting of [2, 2, 1, 3] should return an integer array of {1, 2, 3}. 
         public static int[] UniqueNumbers(params int[] numbers)
         {
-            // Delete the below line when you are writing your code, don't forget to use your
-            // own return statement.
-            return default;
+            if (numbers.Length != numbers.Distinct().Count())
+                Array.Sort(numbers);
+
+            else
+                Array.Sort(numbers);
+                         
+
+           var listNumbers = numbers.ToList();
+            for (var i = 0; i < listNumbers.Count(); i++)
+
+                for (var n = 0; n < listNumbers.Count(); n++)
+
+                    if (listNumbers[i] == listNumbers[n])
+
+                        listNumbers.RemoveAt(n);
+
+            listNumbers.Sort();
```

Comment:
ListNumbers isn't part of the return statement, so everything you do with it is needless.